code/modules/projectiles/guns/energy/special.dm

```diff
@@ -270,12 +270,175 @@
 
 /obj/item/weapon/gun/energy/temperature
 	name = "temperature gun"
-	icon_state = "freezegun"
-	desc = "A gun that changes temperatures."
+	icon = 'icons/obj/gun_temperature.dmi'
+	icon_state = "tempgun_4"
+	item_state = "tempgun_4"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	desc = "A gun that changes the body temperature of its targets."
+	var/temperature = 300
+	var/target_temperature = 300
+	var/e_cost = 100

```

Comment:
this just sets a default so it doesn't runtime if you print one and pull the trigger.